src/KafkaNET.Library/Consumers/PartitionTopicInfo.cs

```diff
@@ -215,9 +212,47 @@ public long FetchOffset
             }
         }
 
+        /// <summary>
+        /// It happens when it gets consumer offset out of range exception. Try to fix the offset.
+        /// </summary>        
+        internal void ResetOffset(long resetOffset)
+        {
+            lock(this.fetchedOffsetLock)
```

Comment:
As I changed FetchedOffset/NextRequestOffset, in case there are other things want to read/write these offsets, better to add a lock.